fe/src/components/UI/molecules/donationModal/DonationModal.tsx

```diff
@@ -0,0 +1,221 @@
+
+interface DonationModalProps {
+  onClick: (
+    volunteerActionTypeId: string,
+    isCompany: boolean,
+    companyName: string,
+    fullName: string,
+    email: string,
+    phoneNumber: string,
+    description: string,
+    isPickup: boolean,
+    address: string,
+    id?: number | null
+  ) => void;
+  onHide: () => void;
+  show: boolean;
+  label: string;
+  actionTypes: VolunteerActionTypeModel[];
+  item?: DonationDTOModel;
+}
+
+const initialValues: any = {
+  isCompany: false,
+  companyName: "",
+  fullName: "",
+  email: "",
+  phoneNumber: "",
+  description: "",
+  isPickup: false,
+  address: "",
+};
+type OptionType = {
+  value: string;
+  label: string;
+};
+export const DonationModal: React.FC<DonationModalProps> = ({
+  onClick,
+  onHide,
+  show = false,
+  item,
+  actionTypes,
+  label,
+}) => {
+  const TypesOptions: OptionType[] = actionTypes.map(
+    (b: VolunteerActionTypeModel) => {
+      return { value: b.id.toString()!, label: b.name! };
+    }
+  );
+  const [selectedType, setSelectedType] = useState(TypesOptions[0]);
+
+  useEffect(() => {
+    setSelectedType(
+      item
+        ? TypesOptions.find(
+            (a) => a.value === item.volunteerActionTypeId.toString()
+          ) ?? TypesOptions[0]
+        : TypesOptions[0]
+    );
+  }, [item]);
+
+  const customStyles = {
+    control: (provided: any) => ({
+      ...provided,
+      border: "1px solid rgba(0, 0, 0, 0.08)",
+      marginBottom: 10,
+      "&:hover": {
+        cursor: "pointer",
+      },
+    }),
+    indicatorSeparator: (provided: any) => ({
+      ...provided,
+      backgroundColor: "rgba(0, 0, 0, 0.1)",
+    }),
+    dropdownIndicator: (provided: any) => ({
+      ...provided,
+      color: "rgba(0, 0, 0, 0.6)",
+    }),
+  };
+
+  const handleOnTypeChange = (e: any) => {
+    setSelectedType(
+      TypesOptions.find((a) => a.value === e.value) ?? TypesOptions[0]
+    );
+  };
+
+  return (
+    <CustomModal show={show} onHide={onHide}>
+      <div>
+        <Formik
+          initialValues={item ? item : initialValues}
+          validationSchema={donationValidationScheme}
+          onSubmit={(values: DonationDTOModel) => {
+            item
+              ? onClick(
+                  selectedType.value,
+                  values.isCompany,
+                  values.companyName,
+                  values.fullName,
+                  values.email,
+                  values.phoneNumber,
+                  values.description,
+                  values.isPickup,
+                  values.address,
+                  item.id
+                )
+              : onClick(
+                  selectedType.value,
+                  values.isCompany,
+                  values.companyName,
+                  values.fullName,
+                  values.email,
+                  values.phoneNumber,
+                  values.description,
+                  values.isPickup,
+                  values.address
```

Comment:
You don't need to do this "if" here. Since you are getting id from item you can pass all other arguments to OnClick, and add item.id directly in onClick function